### PR TITLE
tools: After version 135 the subpackage interdependency is flexible

### PR DESCRIPTION
**Pull request by stefwalter**
The cockpit-ws, cockpit-dashboard, cockpit-bridge and cockpit-system subpackages have a more flexible interdependency after the 135 release. The way they call each other is stable and documented.
    
For the cockpit-bridge and cockpit-system packages we still keep a rather tight dependency for pragmatic reasons. Any later version of cockpit-bridge is acceptable to cockpit-system.

### CODE REVIEW

File: tools/cockpit.spec [hunk 1]
```diff
@@ -272,7 +272,6 @@ cat kdump.list subscriptions.list sosreport.list networkmanager.list selinux.lis
 %package bridge
 Summary: Cockpit bridge server-side component
 Requires: glib-networking
-Requires: polkit
```
Comment (martinpitt) [inline tools/cockpit.spec]: tools/debian/control should then get the same, lowering `Depends: policykit-1` to `Suggests:`.

File: tools/cockpit.spec [hunk 2]
```diff
@@ -364,11 +364,6 @@ test -f %{_bindir}/chcon && chcon -t cockpit_ws_exec_t %{_libexecdir}/cockpit-ss
 
 %package storaged
 Summary: Cockpit user interface for storage, using Storaged
-# Lock bridge dependency due to --with-storaged-iscsi-sessions
-# which uses new less stable /manifests.js request path.
-%if 0%{?rhel}
-Requires: %{name}-bridge >= %{version}-%{release}
-%endif
```
Comment (martinpitt) [inline tools/cockpit.spec]: Agreed, nice cleanup!

File: tools/cockpit.spec [hunk 3]
```diff
@@ -419,7 +419,7 @@ This package contains the Cockpit shell and system configuration interfaces.
 %package tests
 Summary: Tests for Cockpit
 Requires: %{name}-bridge >= %{version}-%{release}
-Requires: %{name}-shell >= %{version}-%{release}
+Requires: %{name}-system >= %{version}-%{release}
```
Comment (martinpitt) [inline tools/cockpit.spec]: The debian -tests package currently doesn't have any dependencies to cockpit* at all, I figure it should grow a `Depends: cockpit-system (>= ${source:Version})` (we can't use `= ${binary:Version}` as c-system is Arch: all).

File: tools/cockpit.spec [hunk 4]
```diff
@@ -327,7 +327,7 @@ The Cockpit components for managing software updates for ostree based systems.
 
 %package pcp
 Summary: Cockpit PCP integration
-Requires: %{name}-bridge = %{version}-%{release}
+Requires: %{name}-bridge > 134
```
Comment (martinpitt) [inline tools/cockpit.spec]: Would it make sense to add a `"requires": {"cockpit": "134"}}` to pkg/pcp/manifest.json.in and use `%required_base` for the dependency instead?

File: tools/cockpit.spec [hunk 5]
```diff
@@ -347,8 +347,8 @@ Cockpit support for reading PCP metrics and loading PCP archives.
 %package dashboard
 Summary: Cockpit SSH remoting and dashboard
 Requires: libssh >= %{libssh_version}
-Requires: cockpit-ws = %{version}-%{release}
-Provides: cockpit-ssh = %{version}-%{release}
+Requires: cockpit-bridge > 134
```
Comment (martinpitt) [inline tools/cockpit.spec]: Likewise, would it make sense  to bump the dependency in pkg/dashboard/manifest.json.in instead and move to `%required_base` here also?

File: tools/cockpit.spec (hunk 4 above)
Comment (dperpeet) [inline tools/cockpit.spec]: I don't think we should artificially bump the requirements in manifest.json - only if the actual content demands it. Anything else is packaging magic/glue on top.

File: tools/cockpit.spec (hunk 5 above)
Comment (dperpeet) [inline tools/cockpit.spec]: I don't think we should artificially bump the requirements in manifest.json - only if the actual content demands it. Anything else is packaging magic/glue on top.

Comment (martinpitt) [inline tools/cockpit.spec]: But having a higher dependency in the downstream packaging  than in the upstream code is completely arbitrary, as we don't have any downstream changes which would change that relationship. Either there is a justification for ≥ 134 and then manifest.json as the one source of truth should reflect that, or there isn't, and then we should just use `%required_base` here without bumping (then it'll come out as `>= 132.x`).

File: tools/cockpit.spec (hunk 4 above)
Comment (stefwalter) [inline tools/cockpit.spec]: Good point. Done.

File: tools/cockpit.spec (hunk 5 above)
Comment (stefwalter) [inline tools/cockpit.spec]: Unfortunately ```required_base``` doesn't yet work for the ```cockpit-system``` package ... which really is a bundle of other packages. I'll leave this manually set in the spec file for now. Do you have ideas on how to make ```gen-spec-dependencies``` work for ```cockpit-system```?

File: tools/cockpit.spec (hunk 3 above)
Comment (stefwalter) [inline tools/cockpit.spec]: Added.

File: tools/cockpit.spec (hunk 4 above)
Comment (stefwalter) [inline tools/cockpit.spec]: The bridge that is declared in the same manifest.json file demands it. So I'd say yes, and bumped.

File: tools/cockpit.spec (hunk 5 above)
Comment (stefwalter) [inline tools/cockpit.spec]: Oh duh. This was about the ```cockpit-dashboard``` not about ```cockpit-system``` ... and yes, the ```manifest.json``` declares the ```"bridges"``` which in turn require that dependency. So updating the ```manifest.json``` makes sense here.

**Comment by martinpitt** [comment]
AFAICS there are two remaining outstanding issues: (1) lower the policykit-1 debian dependency, and (2) using `%required_base` for -dashboard. @stefwalter, want me to push a fixup to this, or want to do that yourself?

This is blocked on landing the cockpit-ssh rework, right?

**Comment by stefwalter** [comment]
> AFAICS there are two remaining outstanding issues: (1) lower the policykit-1 debian dependency, and (2) using %required_base for -dashboard. @stefwalter, want me to push a fixup to this, or want to do that yourself?

You're right. Done.

> This is blocked on landing the cockpit-ssh rework, right?

Indeed.

**Comment by stefwalter** [comment]
Release is imminent removing blocker.

**Comment by martinpitt** [comment]
For {fedora,rhel}-atomic we either need to add cockpit-bridge into the `docker exec rpm -U` command (at the moment this only installs cockpit-dashboard and cockpit-ws) if adding the new dependency is semantically right. Or we need to re-drop the `"+Requires: %{name}-bridge >= %{required_base}"` from the spec.

For container/kubernetes we need to rewrite containers/kubernetes/scripts/install-rpms.sh to install all rpms in one go, not one by one -- like that it always assumes that a sufficiently-versioned older rpms are already installed. That's a larger undertaking, though, not sure if we want to squeeze that in at this point.

**Pull request closed without merging**